random_fields/geostatistical_cpt_interpretation.py

```diff
@@ -0,0 +1,682 @@
+        self.cpt_filepaths: List[Path] = []
+
+        self.interpreter = RobertsonCptInterpretation()
+        self.interpreter.unitweightmethod = UnitWeightMethod.LENGKEEK
+        self.interpreter.shearwavevelocitymethod = ShearWaveVelocityMethod.ZANG
+        self.interpreter.ocrmethod = OCRMethod.MAYNE
+
+        self.data_coords = np.empty((0, 3), dtype=np.float64)
+        self.cpt_locations = np.empty((0, 2), dtype=np.float64)
+        self.rho = np.empty(0)
+        self.vs = np.empty(0)
+        self.G0 = np.empty(0)
+        self.young_modulus = np.empty(0)
+        self.poisson = 0.495  # ToDo: add Poisson ratio see #6
+
+    def read_cpt_data(self, addition: bool = False) -> None:
+        """read_cpt_data Reads in and pre-processes the CPT data
+
+        Args:
+            - addition (bool): If True, reads in the CPT data in addition to the existing data.
+                Defaults to False.
+        """
+
+        if not addition:
+            self.cpt_list = []
+            self.cpt_filepaths = []
+
+        list_of_files = sorted(os.listdir(self.cpt_directory))
+
+        files_type = {'.gef': GefCpt, '.xml': BroXmlCpt}
+
+        for file_ext, cpt_class in files_type.items():
+            # collect gef files
+            gef_files = [f for f in list_of_files if f.endswith(file_ext)]
+
+            for gef_name in gef_files:
+                cpt = cpt_class()
+                cpt.read(self.cpt_directory / gef_name)
+                cpt.pre_process_data()
+
+                self.cpt_list.append(cpt)
+                self.cpt_filepaths.append(self.cpt_directory / gef_name)
+
+        # remove duplicates for the case that both the GEF and XML files are present
+        unique_idx = self.__find_duplicates()
+        self.cpt_list = [self.cpt_list[i] for i in unique_idx]
+        self.cpt_filepaths = [self.cpt_filepaths[i] for i in unique_idx]
+
+    def __find_duplicates(self) -> List[int]:
+        """
+        Finds the duplicates from the CPT data for the case that XML and GEF files are present.
+
+        Returns:
+            - unique_idx (List[int]): list of unique indices
+        """
+        unique_idx = []
+        seen = set()
+        for i, value in enumerate(self.cpt_list):
+            if value.name not in seen:
+                unique_idx.append(i)
+                seen.add(value.name)
+        return unique_idx
+
+    def interpret_cpt_data(self, interpreter: Optional[RobertsonCptInterpretation] = None, v_dim: int = 1) -> None:
+        """
+        Interpret_cpt_data interprets the CPT data based on the provided interpreter settings
+
+        Args:
+            - interpreter (:class:`geolib_plus.robertson_cpt_interpretation.RobertsonCptInterpretation`):
+                Interpretation methods. Defaults to :class:`RobertsonCptInterpretation`.
+            - v_dim (int): index corresponding to vertical. Defaults to 1.
+        """
+
+        if interpreter:
+            self.interpreter = interpreter
```

Comment:
what is the point of this? why whould you use a different interpreter as already defined in the class?